### PR TITLE
Add nested list view for tts page and optional support for single list

### PR DESCRIPTION
**Pull request by AIIX**
- Adds nested list model to TTS Menu for cleaner nested list view 
- Adds optional support for single TTS list menu can be set by image developers,

a-better-tts-selection-menu

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -51,6 +51,51 @@ class PairingMode(str, Enum):
...
+        pass
+
+    def distinguish_models_tts(self, model):
+        """ distinguish the model to a separate list for engines and voices
+        to build a nested list for the GUI """
+        engines_model = []
+
+        for entry in model:
+            if entry["engine"] not in [engine["engine"] for engine in engines_model]:
+                engines_model.append({"engine": entry["engine"], "plugin_name": entry["plugin_name"], "supports_male_voice": False, "supports_female_voice": False, "supports_offline_mode": False, "supports_online_mode": False,  "voiceModel": []})
```
Comment (JarbasAl) [inline __init__.py:68]: this looks wrong, we have a new UI api coming directly from opm, this should be added there instead

please give me a data model with example settings you need, here you are adding a bunch of random stuff such as "supports_XXX" that should be coming directly from OPM instead

File: __init__.py [hunk 2]
```diff
@@ -378,6 +424,9 @@ def _init_setup_options(self):
         if "stt_blacklist" not in self.settings:
             self.settings["stt_blacklist"] = ["ovos-stt-plugin-pocketsphinx"]
 
+        if "single_tts_list" not in self.settings:
+            self.settings["single_tts_list"] = False
```
Comment (JarbasAl) [inline __init__.py:428]: default this to True please, this new list view is a bigscreen specific requirement and does not match the official OPM api, at a later point we may decide to make this a default but for now we need to keep compat

File: __init__.py [hunk 3]
```diff
@@ -51,6 +51,51 @@ class PairingMode(str, Enum):
...
+        engines_model = []
+
+        for entry in model:
+            if entry["engine"] not in [engine["engine"] for engine in engines_model]:
+                engines_model.append({"engine": entry["engine"], "plugin_name": entry["plugin_name"], "supports_male_voice": False, "supports_female_voice": False, "supports_offline_mode": False, "supports_online_mode": False,  "voiceModel": []})
+
+        for entry in model:
+            for engine in engines_model:
+                if entry["engine"] == engine["engine"]:
+                    engine["voiceModel"].append(entry)
```
Comment (JarbasAl) [inline __init__.py:73]: nitpick, in python we dont use CamelCase, change to voice_model for consistency across all code base and projects

**Review by JarbasAl — CHANGES_REQUESTED** [review]
these PRs can be updated for your needs, please rebase on them and address feedback above about needed data model, they need to be merged first and version pinned in requirements.txt of this skill

https://github.com/OpenVoiceOS/ovos-plugin-manager/pull/85 + https://github.com/OpenVoiceOS/skill-ovos-setup/pull/57

File: __init__.py (hunk 2 above)
Comment (AIIX) [reply __init__.py:428]: This is not bigscreen specific this is for manjaro ovos images, any other image can decide how they want their tts menu workflow to look

File: __init__.py (hunk 1 above)
Comment (AIIX) [reply __init__.py:68]: This is not random stuff, this is directly coming from the api in the setup skill

It's only evaluating:

{"plugin_name": plugin_display_name, "display_name": voice.get("display_name", " "), "gender": voice.get("gender", " "), "offline": voice.get("offline", False), "lang": lang, 'engine': engine}

To build a new list that will be displayed as the engine models list that simply states after evaluating the voices in the engine whether the engine supports male or female or none gender voices and whether the engine supports both online and offline voices

This does not add anything new, this is a QML GUI listview model

File: __init__.py (hunk 3 above)
Comment (AIIX) [reply __init__.py:73]: There is no rule for this nitpick and it seems quite like forcing a coding style

**Comment by AIIX** [comment]
This QML listview model has nothing to do with the key model coming from OPM and can easily be adapted to whatever model OPM provides, so it does not need to live or be provided by OPM at all

It only consumes the same data model that OPM is anyways providing

**Pull request closed without merging**